### PR TITLE
bpo-28859: Fix return value of ismount on Windows

### PR DESCRIPTION
**Pull request by digglife**
When a path doest not acutally exist, `ismount` should return False
instead of raising FileNotFountError.

https://bugs.python.org/issue28859

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@digglife

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Lib/ntpath.py [hunk 1]
```diff
@@ -272,6 +272,8 @@ def ismount(path):
         return True
 
     if _getvolumepathname:
+        if not lexists(path):
```
Comment (lazka) [inline Lib/ntpath.py]: What if the path gets removed right after this line executes?

Comment (digglife) [reply Lib/ntpath.py]: Oops! So I almost introduced a potential bug for fixing a bug...

How about explicitly catching the `FileNotFoundError` that could be thrown by `_getvolumepathname`?

```suggestion
        try:
            return path.rstrip(seps) == _getvolumepathname(path).rstrip(seps)
        except FileNotFoundError:
            return False
```

Comment (digglife) [reply Lib/ntpath.py]: Hi Chris @lazka

Could you please review my update?

Comment (lazka) [reply Lib/ntpath.py]: lgtm

File: Lib/ntpath.py [hunk 2]
```diff
@@ -272,7 +272,10 @@ def ismount(path):
         return True
 
     if _getvolumepathname:
-        return path.rstrip(seps) == _getvolumepathname(path).rstrip(seps)
+        try:
+            return path.rstrip(seps) == _getvolumepathname(path).rstrip(seps)
+        except FileNotFoundError:
```
Comment (zooba) [inline Lib/ntpath.py]: Let's handle `OSError` here, like in the other functions.

File: Lib/test/test_ntpath.py [hunk 3]
```diff
@@ -661,6 +661,8 @@ def test_ismount(self):
         self.assertTrue(ntpath.ismount(b"\\\\.\\c:\\"))
         self.assertTrue(ntpath.ismount(b"\\\\.\\C:\\"))
 
+        self.assertFalse(ntpath.ismount("C:\\NotExist"))
```
Comment (zooba) [inline Lib/test/test_ntpath.py]: The drive also needs to not exist - this check already passes on most machines without your fix.

I suggest something like:
```python
for drive in "DEFGHIJKLMNOPQRSTUVWXYZ":
    if not ntpath.exists(drive + ":\\"):
        self.assertFalse(ntpath.ismount(drive + ":\\NotExist"))
        break
else:
    if support.verbose:
        print("No missing drive found to test 'ismount'")
```

Comment (digglife) [reply Lib/test/test_ntpath.py]: This is smart and reliable.

**Comment by digglife** [comment]
@zooba 

Thanks for your review. I updated the code as your suggestion. Could you please take a look?

**Comment by digglife** [comment]
Wow, it's been a year since I sent this. Is there anything I can do to push it?

**Pull request closed without merging**